test/integration/smoke/test_primary_storage.py

```diff
@@ -336,6 +335,7 @@ def test_01_add_primary_storage_disabled_host(self):
                     serviceofferingid=service_offering.id
                 )
                 self.cleanup.append(self.virtual_machine)
+                self.cleanup.append(service_offering)
```

Comment:
should not be happening here. You are probably doing this because the teardown implementation is no good